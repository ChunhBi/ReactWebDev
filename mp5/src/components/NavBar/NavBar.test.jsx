import renderer from 'react-test-renderer'
import NavBar from './NavBar.jsx';
import {test, expect, vi} from 'vitest'

test('The NavBar component should render', () => {
    const view = renderer
        .create(<NavBar/>)
        .toJSON();

    expect(view).toMatchSnapshot();
});

test('The NavBar component should render with a title', () => {
    const title = 'Test application';
    const view = renderer.create(<NavBar title={title}/>);

    // console.log(view.root.findByType('h2').children[0]);
    expect(view.root.findByType('h2').children[0]).toBe(title);
});

test('The NavBar component should respond to button clicks', () => {
    const mockFunction = vi.fn();
    const view = renderer.create(<NavBar goBack={mockFunction} openForm={mockFunction}/>);

    view.root.findByProps({id: 'backbtn'}).props.onClick();
    expect(mockFunction).toHaveBeenCalled();

    view.root.findByProps({id: 'openbtn'}).props.onClick();
    expect(mockFunction).toHaveBeenCalledTimes(2);
});
