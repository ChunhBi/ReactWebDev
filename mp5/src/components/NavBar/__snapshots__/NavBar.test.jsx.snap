// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The NavBar component should render 1`] = `
<div
  className="sc-guDLey biTsvu"
>
  <h2
    className="sc-dmyCSP dmoMIl"
    id="title"
  />
</div>
`;
